src/App.tsx

```diff
@@ -3,33 +3,58 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+  filteredMovies: Movie[],
+};
 
-export class App extends React.Component<{}, State> {
-  state: State = {};
+export class App extends React.PureComponent<{}, State> {
+  state: State = {
+    query: '',
+    filteredMovies: [...moviesFromServer],
+  };
+
+  changeInput = (someText: string) => {
+    this.setState({ query: someText });
+
+    this.getFilteredMovies();
+  };
+
+  getFilteredMovies = () => {
+    const { query } = this.state;
+
+    this.setState({
+      filteredMovies: moviesFromServer.filter(movie => movie.title.toLowerCase()
+        .includes(query.toLowerCase())
```

Comment:
no need to do `query.toLowerCase()` on every iteration, you can do it before filtering